src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ProducerConsumer.cs

```diff
@@ -263,22 +263,38 @@ await args.produceItems(source, callback, args.args, cancellationToken).Configur
     /// Version of <see cref="RunChannelAsync"/> when caller the prefers to just push all the results into a channel
     /// that it receives in the return value to process asynchronously.
     /// </summary>
-    public static async IAsyncEnumerable<TItem> RunAsync<TArgs>(
+    public static IAsyncEnumerable<TItem> RunAsync<TArgs>(
         Func<Action<TItem>, TArgs, CancellationToken, Task> produceItems,
         TArgs args,
-        [EnumeratorCancellation] CancellationToken cancellationToken)
+        CancellationToken cancellationToken)
     {
-        var channelReader = await RunChannelAsync(
-            // We're the only reader (in the foreach loop below).  So we can use the single reader options.
-            ProducerConsumerOptions.SingleReaderOptions,
-            produceItems,
-            // Trivially grab the reader and return it.  We don't need to do any processing of the values, as the
-            // callers just wants them as is.
-            consumeItems: static (reader, _, _) => Task.FromResult(reader),
-            args, cancellationToken).ConfigureAwait(false);
-
-        await foreach (var item in channelReader.ReadAllAsync(cancellationToken))
```

Comment:
could you pls explain where the buffering happens before? It's not obvious to me from looking at the change. Thanks :)

Reply:
Awaiting RunChannelAsync Waits for all working to complete. 

But in your case, that can't happen because it blocks producing items. This happens prior to even yielding any values out:-(